app/controllers/search_controller.rb

```diff
@@ -12,7 +12,8 @@ def index
     if search_params.no_search? && params[:format] != "json"
       render action: 'no_search_term' and return
     end
-    search_response = SearchAPI.new(search_params).search
+    variant = search_match_length_variant.variant_name
+    search_response = SearchAPI.new(search_params, match_length_variant: variant).search
```

Comment:
I wonder whether it would be worth making this parameter name match the A/B test name? It would give a simple mapping between the A/B cookie, header and API parameter. It would probably be easier to understand when debugging.